server/src/test/java/org/opensearch/common/breaker/ResponseLimitSettingsTests.java

```diff
@@ -0,0 +1,204 @@
+import org.opensearch.cluster.routing.TestShardRouting;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.index.Index;
+import org.opensearch.core.index.shard.ShardId;
+import org.opensearch.test.OpenSearchTestCase;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.opensearch.common.breaker.ResponseLimitSettings.CAT_INDICES_RESPONSE_LIMIT_SETTING;
+import static org.opensearch.common.breaker.ResponseLimitSettings.CAT_SEGMENTS_RESPONSE_LIMIT_SETTING;
+import static org.opensearch.common.breaker.ResponseLimitSettings.CAT_SHARDS_RESPONSE_LIMIT_SETTING;
+
+public class ResponseLimitSettingsTests extends OpenSearchTestCase {
+
+    public void testIsResponseLimitBreachedForNullMetadataExpectNotBreached() {
+        final Settings settings = Settings.builder().build();
+        final ClusterSettings clusterSettings = new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        final ResponseLimitSettings responseLimitSettings = new ResponseLimitSettings(clusterSettings, settings);
+        final boolean breached = ResponseLimitSettings.isResponseLimitBreached(
+            (Metadata) null,
+            ResponseLimitSettings.LimitEntity.INDICES,
+            0
+        );
+        assertFalse(breached);
+    }
+
+    public void testIsResponseLimitBreachedForNullRoutingTableExpectNotBreached() {
+        final Settings settings = Settings.builder().build();
+        final ClusterSettings clusterSettings = new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        final ResponseLimitSettings responseLimitSettings = new ResponseLimitSettings(clusterSettings, settings);
+        final boolean breached = ResponseLimitSettings.isResponseLimitBreached(
+            (RoutingTable) null,
+            ResponseLimitSettings.LimitEntity.INDICES,
+            0
+        );
+        assertFalse(breached);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testIsResponseLimitBreachedForNullLimitEntityWithRoutingTableExpectException() {
+        final ClusterState clusterState = buildClusterState("test-index-1", "test-index-2", "test-index-3");
+        ResponseLimitSettings.isResponseLimitBreached(clusterState.getRoutingTable(), null, 4);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testIsResponseLimitBreachedForNullLimitEntityWithMetadataExpectException() {
+        final ClusterState clusterState = buildClusterState("test-index-1", "test-index-2", "test-index-3");
+        ResponseLimitSettings.isResponseLimitBreached(clusterState.getMetadata(), null, 4);
+    }
+
+    public void testIsResponseLimitBreachedForCatIndicesWithSettingDisabledExpectNotBreached() {
+        // Don't enable limit
+        final Settings settings = Settings.builder().put(CAT_INDICES_RESPONSE_LIMIT_SETTING.getKey(), -1).build();
```

Comment:
Can we add a test to pass null value for this setting and verify it is not enforced ?